src/FTPClientWrapperSSH.cpp

```diff
@@ -553,47 +552,19 @@ int FTPClientWrapperSSH::authenticate_key(ssh_session session) {
 	}
 
 	char * keyfile = SU::TCharToCP(m_keyFile, CP_ACP);
-	ssh_private_key privkey = NULL;
-	if (m_passphrase[0] == 0)	//in case the passphrase is empty, use NULL instead
-		privkey = privatekey_from_file(session, keyfile, 0, NULL);
-	else
-		privkey = privatekey_from_file(session, keyfile, 0, m_passphrase);
+	ssh_key privkey = NULL;
+	//in case the passphrase is empty, use NULL instead
+	rc = ssh_pki_import_privkey_file(keyfile, m_passphrase[0] ? m_passphrase : NULL, NULL, NULL, &privkey);

```

Comment:
I removed the PathFileExists check and verified that it works as expected for local files, files on removable devices, and files on remote network shares (SMB). I also verified for invalid (non-existing) paths, for all three cases, that the following message is logged and no crash occurs.

[SFTP] Invalid private key file

Let me know if you would like me to commit this change.